Composer/packages/lib/bot-deploy/src/botProjectDeploy.ts

```diff
@@ -595,7 +595,10 @@ export class BotProjectDeploy {
         message: 'Publish To Azure Success!',
       });
     } catch (error) {
-      console.log(error);
+      this.logger({
+        status: BotProjectDeployLoggerType.DEPLOY_ERROR,
+        message: JSON.stringify(error, Object.getOwnPropertyNames(error)),
```

Comment:
What is the intention of the second argument here? If I recall the api, it should be a replacer function, but `Object.getOwnPropertyNames` returns an array.

Reply:
> What is the intention of the second argument here? If I recall the api, it should be a replacer function, but `Object.getOwnPropertyNames` returns an array.

I'm trying to log down the native javascript exception error to json string, and referring to this answer on stackoverflow: https://stackoverflow.com/a/26199752

Reply:
Ah ok. I learned something new about `JSON.stringify`!